language/move-prover/src/prelude.bpl

```diff
@@ -1131,6 +1111,40 @@ procedure {:inline 1} Signature_ed25519_threshold_verify(bitmap: Value, signatur
 
 // TODO: implement the below methods
 
-procedure {:inline 1} $LCS_to_bytes(ta: TypeValue, v: Reference) returns (res: Value) {
-    assert false; // $LCS_to_bytes not implemented
-}
+// ==================================================================================
+// Native LCS::serialize
+
+// native public fun serialize<MoveValue>(v: &MoveValue): vector<u8>;
+
+// Serialize is modeled as (almost) uninterpreted function, with an additional
+// axiom to say it's an injection.
+
+function $serialize(ta: TypeValue, v: Value): Value;
+
+// This says that $serialize respects isEquals (substitution property)
+// Without this, Boogie will get false positives where v1, v2 differ at invalid
+// indices.
+axiom (forall ta: TypeValue ::
```

Comment:
Just wondering: does this need to be a nested quantor?

Reply:
I just did that because I'm not sure how general Boogie quantifier syntax is.  I'm guessing that it doesn't matter to Boogie, and it seems to work in practice.